packages/core/src/routes/profile/index.ts

```diff
@@ -30,29 +34,35 @@ export default function profileRoutes<T extends UserRouter>(
       body: z.object({
         name: z.string().nullable().optional(),
         avatar: z.string().url().nullable().optional(),
+        username: z.string().regex(usernameRegEx).optional(),
       }),
       response: z.object({
         name: z.string().nullable().optional(),
         avatar: z.string().nullable().optional(),
+        username: z.string().optional(),
       }),
-      status: [200, 400],
+      status: [200, 400, 422],
     }),
     async (ctx, next) => {
       const { id: userId, scopes } = ctx.auth;
-      const {
-        body: { name, avatar },
-      } = ctx.guard;
+      const { body } = ctx.guard;
+      const { name, avatar, username } = body;
 
       assertThat(scopes.has(UserScope.Profile), 'auth.unauthorized');
 
-      const updatedUser = await updateUserById(userId, { name, avatar });
+      if (username !== undefined) {
+        await checkIdentifierCollision({ username }, userId);
+      }
+
+      const updatedUser = await updateUserById(userId, { name, avatar, username });
```

Comment:
Can we delete a username using this PATCH API?

Reply:
We can not, the `usernameRegEx` can ensure that.